### PR TITLE
Find out correct system memory when run in a container.

### PR DESCRIPTION
**Pull request by kainwen**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@kainwen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@kainwen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@kainwen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@kainwen Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=2779)!

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 1]
```diff
@@ -758,16 +771,30 @@ ResGroupOps_GetCpuCores(void)
...
+	getCgMemoryInfo(&cgram, &cgmemsw);
+	ram = Min(ram, cgram);
+	/*
+	 * In the case that total ram and swap read from sysinfo is larger than
+	 * from cgroup, ram and swap must both be limited, otherwise swap must
+	 * not be limited(we can safely use the value from sysinfo as swap size).
+	 */
+	if (cgmemsw < memsw)
+		swap = cgmemsw - ram;
 	total = swap + ram * overcommitRatio / 100;
```
Comment (xiong-gang) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: is the still correct after we calibrated the swap?

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 2]
```diff
@@ -758,16 +771,35 @@ ResGroupOps_GetCpuCores(void)
...
+	getCgMemoryInfo(&cgram, &cgmemsw);
+	ram = Min(ram, cgram);
+	/*
+	 * In the case that total ram and swap read from sysinfo is larger than
+	 * from cgroup, ram and swap must both be limited, otherwise swap must
+	 * not be limited(we can safely use the value from sysinfo as swap size).
+	 */
+	if (cgmemsw < memsw)
+		swap = cgmemsw - ram;
+	/* If it is in container, the total memory is limited by both the total
```
Comment (xiong-gang) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: add a new line here please

**Pull request merged**